php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -163,6 +163,14 @@ protected function writeOneof($number, $value)
         $oneof_field->setNumber($number);
     }
 
+    protected function whichOneof($oneof_name)
+    {
```

Comment:
whichOneof is for implementation only. It's used to implement API in generated code for oneof.
For proto:
message Foo {
  oneof my_oneof {
    int32 a = 1;
    int32 b = 1;
  }
}
In generated code, message Foo will have the method "getMyOneof", which returns the name of the field that is currently set (e.g., a or b).